src/instructlab/model/train.py

```diff
@@ -279,9 +303,67 @@
 @click.option(
     "--legacy",
     is_flag=True,
-    default=False,
     help="if true, enables the legacy linux training codepath from release 0.17.0 and prior.",
 )
+@click.option(
+    "--strategy",
+    type=click.Choice(["lab-multiphase"], case_sensitive=False),
+    show_default=True,
+    help="If chosen, will run the selected training strategy instead of a single training run.",
+)
+@click.option(
+    "--phased-base-dir",
+    type=clickpath_setup(is_dir=True),
+    default=lambda: DEFAULTS.PHASED_DIR,
+    help="Base directory for organization of end-to-end intermediate outputs.",
+)
+@click.option(
+    "--phased-phase1-data",
+    type=clickpath_setup(is_dir=False),
+    help="Path to .jsonl file that will be used for the first phase of end-to-end training.",
+)
+@click.option(
+    "--phased-phase1-num-epochs",
+    cls=clickext.ConfigOption,
+    type=int,
+    help="Number of epochs to run for first phase of end-to-end training.",
+)
+@click.option(
+    "--phased-phase1-samples-per-save",
+    cls=clickext.ConfigOption,
+    type=int,
+    help="Number of samples to train on between saves for first phase of end-to-end training.",
+)
+@click.option(
+    "--phased-phase2-data",
+    type=clickpath_setup(is_dir=False),
+    help="Path to .jsonl file that will be used for the second phase of end-to-end training.",
+)
+@click.option(
+    "--phased-phase2-num-epochs",
+    cls=clickext.ConfigOption,
+    type=int,
+    help="Number of epochs to run for second phase of end-to-end training.",
+)
+@click.option(
+    "--phased-phase2-samples-per-save",
+    cls=clickext.ConfigOption,
+    type=int,
+    help="Number of samples to train on between saves for second phase of end-to-end training.",
+)
+@click.option(
+    "--phased-mt-bench-judge",
+    # type=clickpath_setup(is_dir=True), # want this in the future, can't guarantee it exists so can't enforce it this way.
```

Comment:
(Curious) Why can't we expect all the input (data and models) accessible before starting the run?

Reply:
They might not exist if we're not using the phased strategy. i.e. the user doesn't need prometheus if they're just trying out training.